tests/scanners/test.javascript.js

```diff
@@ -227,6 +228,45 @@ describe('JavaScript Scanner', function() {
     assert(ok);
   });
 
+  it('should pass addon metadata to rules', () => {
+    var fakeRules = { metadata_not_passed: () => {} };
+
+    sinon.stub(fakeRules, 'metadata_not_passed', (context) => {
+      return {
+        Identifier: (node) => {
+          var metadata = context.settings.addonMetadata;
+
+          if (!metadata || typeof metadata.guid === 'undefined') {
```

Comment:
GUID is guaranteed here as it's only running this rule in this test, where we pass in a metadata object with a GUID.

I guess I could either check for an object (less explicit) or be even _more_ explicit and test for the exact GUID value. I'll do the latter if that's cool.